src/libs/actions/IOU.ts

```diff
@@ -217,6 +217,32 @@ type MoneyRequestInformationParams = {
     existingTransaction?: OnyxEntry<OnyxTypes.Transaction>;
 };
 
+type MoneyRequestOptimisticParams = {
+    chatReport: OnyxTypes.OnyxInputOrEntry<OnyxTypes.Report>;
+    iouReport: OnyxTypes.Report;
+    transaction: OnyxTypes.Transaction;
+    reportPreviewAction: ReportAction;
+    chatCreatedAction: OptimisticCreatedReportAction;
+    iouCreatedAction: OptimisticCreatedReportAction;
+    iouAction: OptimisticIOUReportAction;
+    transactionThreadReport: OptimisticChatReport | null;
+    transactionThreadCreatedReportAction: OptimisticCreatedReportAction | null;
+    optimisticPersonalDetailListAction?: OnyxTypes.PersonalDetailsList;
+    optimisticPolicyRecentlyUsedCategories?: string[];
+    optimisticPolicyRecentlyUsedTags?: OnyxTypes.RecentlyUsedTags;
+    optimisticNextStep?: OnyxTypes.ReportNextStep | null;
+    optimisticRecentlyUsedCurrencies?: string[];
+};
```

Comment:
Can we please organize this a bit more? Let's group the fields into reports, actions, transactionParams, and recentlyUsed. Anything else that doesn't fit can be a top level field. Also, let's please remove the optimistic prefix since that's redundant.

How about this? I think that would make it really easy and clear to access each part with dot notation, such as `iou.report`, `policyRecentlyUsed.categories`, etc.
```ts
type MoneyRequestOptimisticParams = {
    chat: {
        report: OnyxTypes.OnyxInputOrEntry<OnyxTypes.Report>;
        createdAction: OptimisticCreatedReportAction;
    };
    iou: {
        action: OptimisticIOUReportAction;
        createdAction: OptimisticCreatedReportAction;
        reportPreviewAction: ReportAction;
        report: OnyxTypes.Report;
    };
    transactionParams: {
        transaction: OnyxTypes.Transaction;
        threadCreatedReportAction: OptimisticCreatedReportAction | null;
    };
    policyRecentlyUsed: {
        categories?: string[];
        tags?: OnyxTypes.RecentlyUsedTags;
        currencies?: string[];
    };
    personalDetailListAction?: OnyxTypes.PersonalDetailsList;
    nextStep?: OnyxTypes.ReportNextStep | null;
};
```

Reply:
@neil-marcellini Agree, I updated.

cc @dukenv0307